cmd/desync/extract_test.go

```diff
@@ -154,3 +154,44 @@ func TestExtractWithInvalidSeeds(t *testing.T) {
 		})
 	}
 }
+
+func TestExtractWithNonStaticSeeds(t *testing.T) {
+	outDir, err := ioutil.TempDir("", "")
+	require.NoError(t, err)
+	defer os.RemoveAll(outDir)
+	out := filepath.Join(outDir, "out")
+
+	for _, test := range []struct {
+		name   string
+		args   []string
+		index  string
+		output string
+	}{
+		{"extract with seed that becomes invalid while extracting",
+			[]string{"--store", "testdata/blob2.store", "--seed", "testdata/blob2_corrupted.caibx", "--regenerate-invalid-seeds"}, "testdata/blob2.caibx", out},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			// Assume the seed to be okay while validating the plan
+			// When targeting at least Go 1.17, we can replace the following with "t.Setenv()"
+			err := os.Setenv("DESYNC_MOCK_VALID_PLAN", "1")
+			t.Cleanup(func() { os.Unsetenv("DESYNC_MOCK_VALID_PLAN") })
```

Comment:
Go 1.17 introduced scoped env vars, so these two lines could be replaced with `t.Setenv("DESYNC_MOCK_VALID_PLAN", "1")`

Github VMs seems to include Go 1.17 now https://github.com/actions/virtual-environments/issues/5280 but in `go.mod` the minimum Go version is the 1.15.
Probably not worth it to bump the Go version for just this minor improvement?

Reply:
Yes, it's definitely time to bump the min version to 1.17 since 1.18 is out already. Should also update the dependencies at that time.

Reply:
Now that I don't use the environment variable anymore, we can probably bump the min version in a separate PR.